crates/sui/src/ptb/ptb.rs

```diff
@@ -339,22 +340,18 @@ impl PTB {
 
         // Build the PTB
         let mut parser = PTBParser::new();
-        for command in commands {
+        for command in commands.clone() {
             parser.parse(command.1);
         }
 
         let (parsed, errors) = parser.finish();
 
         if !errors.is_empty() {
-            let errs: Vec<_> = errors
-                .iter()
-                .enumerate()
-                .map(|(i, e)| format!("[{i}]: {e}"))
-                .collect();
-            anyhow::bail!(
-                "Encountered errors when parsing the PTB\n{}",
-                errs.join("\n")
-            );
+            let rendered = render_errors(commands, errors);
+            for e in rendered.iter() {
+                println!("{:?}", e);
+            }
+            anyhow::bail!("Encountered errors when parsing the PTB",);
```

Comment:
I sense that this should go before the for loop?

Reply:
I went with it after so it would look something like:

```
Error 1
...
Error n

Encountered errors when parsing the PTB
```

Which makes it easier to spot that there was an error (since the immediate line before the next command prompt says there's an error). But I can put it up-top if that's preferable :)